app/controllers/mixins/actions/vm_actions/ownership.rb

```diff
@@ -96,47 +84,44 @@ def ownership(ownership_items = [])
           end
         end
 
-        def filter_ownership_items(klass, ownership_items)
-          @origin_ownership_items = ownership_items
-          @ownershipitems ||= begin
-            ownership_scope = klass.where(:id => ownership_items)
-            ownership_scope = ownership_scope.with_ownership if klass.respond_to?(:with_ownership)
-            Rbac.filtered(ownership_scope.order(:name))
-          end
-        end
-
         def build_ownership_info(ownership_items)
           @edit ||= {}
           klass = get_class_from_controller_param(params[:controller])
-          record = klass.find(ownership_items[0])
-          user = record.evm_owner if ownership_items.length == 1
-          @user = user ? user.id.to_s : nil
+          ownershipitems = find_records_with_rbac(klass, ownership_items).sort_by(&:name)
+          record = ownershipitems.first
```

Comment:
This part can be extracted in new method, you are using the same code in `build_ownership_hash`

```ruby
          ownershipitems = find_records_with_rbac(klass, ownership_items).sort_by(&:name)
          record = ownershipitems.first

          if ownership_items.length > 1
            @user = @group = 'dont-change'
          else
            @user = record.evm_owner&.id&.to_s
            @group = record.miq_group&.id&.to_s
          end
```